xdsl/transforms/convert_onnx_to_linalg.py

```diff
@@ -62,24 +62,23 @@ def match_and_rewrite(self, add: onnx.Add, rewriter: PatternRewriter, /):
 class ReluOpLowering(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, relu: onnx.Relu, rewriter: PatternRewriter, /):
-        body = Region(Block(arg_types=(f64, f64)))
-        affine_map = AffineMapAttr(AffineMap.from_callable(lambda d0, d1: (d0, d1)))
+        operand = relu.operand.type
+        assert isinstance(operand, TensorType)
+        operand = cast(TensorType[Attribute], operand)
+        operand_rank = len(operand.get_shape())
+        element_type = cast(FloatAttr[Any], operand.element_type)
+        body = Region(Block(arg_types=(element_type, element_type)))
+        affine_map = AffineMapAttr(AffineMap.identity(operand_rank))
         rewriter.replace_matched_op(
             (
                 empty := tensor.EmptyOp((), relu.res.type),
-                zero := arith.Constant(FloatAttr(0, f64)),
+                zero := arith.Constant(FloatAttr(0.0, element_type)),
```

Comment:
ah I see, that's my bad, here's what the line should be:
```suggestion
                zero := arith.Constant(FloatAttr(0.0, cast(AnyFloat, operand.element_type))),
```

Reply:
I mean this cast but above is probably better